spec/support/with_undefine_mock_controller.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+RSpec.shared_context 'with undefine mock controller' do
+  let(:undefine_mock_controller) do
+    # NB: it is important to undefine the mock controller after the test, otherwise other rspecs may fail
+    Object.send(:remove_const, :MockController) if Object.const_defined?(:MockController)
+    Rails.application.reload_routes!
+  end
+end
```

Comment:
See above comments - this feels a little funky, would be nice if it could be removed?